plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import fetch from 'node-fetch';
+
+
+const dataBaseLocation = `https://raw.githubusercontent.com/google/blockly/
+develop/scripts/migration/renamings.js`;
+
+/**
+ * A temporary top level function that can be called for manual testing.
+ * @param {string} currVersion The version to migrate from.
+ * @param {string} newVersion The version to migrate to.
+ * @param {!Array<string>} files The names files to apply the renamings in.
+ */
+export async function tempTopLevel(currVersion, newVersion, files) {
+  const database = await getDatabase();
+  // TODO: Fix this to actually get the contents of the files.
+  doRenamings(database, currVersion, newVersion, files);
+}
+
+/**
+ * Gets the database of renames.
+ * @return {!Promise<Object>} The database of renames as an object.
+ */
+async function getDatabase() {
+  const response = await fetch(dataBaseLocation);
+  const body = await response.text();
+  console.log(body);
+  // return JSON5.parse(body);
+}
+
+/**
+ * Upgrades pieces of the Blockly API that have been renamed by modifying
+ * developers' local files.
+ * @param {!Object} database The database of renames.
+ * @param {string} currVersion The version to migrate from.
+ * @param {string} newVersion The version to migrate to.
+ * @param {!Array<string>} strings The strings to apply the renamings in.
+ * @return {!Array<string>} The strings with renamings applied.
+ */
+export function doRenamings(database, currVersion, newVersion, strings) {
```

Comment:
I separated this part of the algorithm out into an exported function to make the system easier to test.

Reply:
Yes, this looks good, at the very least as a place to start.